subcommands/targets/show.go

```diff
@@ -163,8 +171,22 @@ func doShowComposeApp(cmd *cobra.Command, args []string) {
 }
 
 func getTargets(factory string, version string) (map[string]string, map[string]client.TufCustom) {
-	targets, err := api.TargetsList(factory, version)
-	subcommands.DieNotNil(err)
+	var targets data.Files
+	if _, err := strconv.Atoi(version); err == nil {
+		logrus.Debug("Looking up targets by version")
+		targets, err = api.TargetsList(factory, version)
+		subcommands.DieNotNil(err)
+	} else {
+		logrus.Debug("Looking up target by name")
+		targets, err = api.TargetsList(factory)
+		subcommands.DieNotNil(err)
+		target, ok := targets[version]
```

Comment:
There is an api `GET /<string:factory>/targets/<string:target>/` which returns a single target by its ID.
Thus, there is no need to fetch entire list and do a lookup locally.

Reply:
https://github.com/foundriesio/fioctl/pull/109/commits/553ab17170e31726a938ad5c920191deb4d1f4ac